src/components/withParentReportAction.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+import ONYXKEYS from '../ONYXKEYS';
+import reportActionPropTypes from '../pages/home/report/reportActionPropTypes';
+import reportPropTypes from '../pages/reportPropTypes';
+
+const withParentReportActionPropTypes = {
+    parentReportAction: PropTypes.shape(reportActionPropTypes),
+};
+
+const withParentReportActionDefaultProps = {
+    parentReportAction: {},
+};
+
+export default function (WrappedComponent) {
+    const propTypes = {
+        forwardedRef: PropTypes.oneOfType([
+            PropTypes.func,
+            PropTypes.shape({current: PropTypes.instanceOf(React.Component)}),
+        ]),
+
+        /** All Report actions for the parent report */
+        parentReportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+
+        /** Thread (child) report */
+        report: reportPropTypes,
+    };
+    const defaultProps = {
+        forwardedRef: undefined,
+        report: null,
+        parentReportActions: null,
+    };
+
+    const WithParentReportAction = (props) => {
+        const parentReportActionID = props.report ? `${lodashGet(props.report, 'parentReportActionID', '')}` : '';
+        const parentReportAction = parentReportActionID ? lodashGet(props.parentReportActions, parentReportActionID, {}) : {};
```

Comment:
I'm struggling to test this since updating auth.

Reply:
Not sure why this is blocker, you can still set parentReportID via an onyx merge